packages/headless/src/features/search-parameters/search-parameter-insight-analytics-actions.ts

```diff
@@ -0,0 +1,145 @@
+import {SearchParameters} from './search-parameter-actions';
+import {DateRangeRequest, NumericRangeRequest} from '../../controllers';
+import {logResultsSort} from '../sort-criteria/sort-criteria-insight-analytics-actions';
+import {logInsightInterfaceChange} from '../analytics/analytics-actions';
+import {
+  logFacetClearAll,
+  logFacetDeselect,
+  logFacetSelect,
+} from '../facets/facet-set/facet-set-insight-analytics-actions';
+import {logSearchboxSubmit} from '../query/query-insight-analytics-actions';
+
+export function logParametersChange(
+  previousParameters: SearchParameters,
+  newParameters: SearchParameters
+) {
+  if (previousParameters.q !== newParameters.q) {
+    return logSearchboxSubmit();
+  }
+
+  if (previousParameters.sortCriteria !== newParameters.sortCriteria) {
+    return logResultsSort();
+  }
+
+  if (areFacetParamsEqual(previousParameters.f, newParameters.f)) {
```

Comment:
Shouldn't we log the facet analytics action when parameters are different instead?

If it is a mistake, `cf`, `nf` and `df` have to be fixed as well.

Reply:
I think the name is no good 😭 

´´´
function areFacetParamsEqual(
  previousFacetParams: AnyFacetParameters = {},
  newFacetParams: AnyFacetParameters = {}
) {
  return JSON.stringify(previousFacetParams) !== JSON.stringify(newFacetParams);
}
´´´